### ISSUE TITLE
Add builds for ppc64le, aarch64

### ISSUE THREAD

**Issue by mattip** [conda-forge/ray-packages-feedstock#75]
There have been some requests on the ray issue tracker for wider support across architectures. It would be nice to supply them via conda

xref ray-project/ray#7476, ray-project/ray#4309, ray-project/ray#21786

**Comment by h-vetinari** [comment]
According to the migrator status, this is also blocked on gym. I guess we really have to tackle pygame eventually (ale-py we could live without)

**Comment by mattip** [comment]
I think upstream will move to [gymnasium (a friendly fork)](https://github.com/Farama-Foundation/Gymnasium) instead of gym at some point. That will come with a [whole new set of dependencies](https://github.com/Farama-Foundation/Gymnasium/blob/main/requirements.txt), including pygame and ale-py.

**Comment by h-vetinari** [comment]
> That will come with a [whole new set of dependencies](https://github.com/Farama-Foundation/Gymnasium/blob/main/requirements.txt), including pygame and ale-py.

Aside from those two we should be good. But definitely timely to start looking at pygame again. 😅

**Comment by richardliaw** [comment]
Sorry for jumping on this so late. Can we actually first make progress on ARM support mainly for Ray Core, ignoring Gym/RLlib?

**Comment by mattip** [comment]
I am not sure how that would work. There would not be a ray-rllib package for aarch64?

On the other hand, support for ale-py was already merged, and support for pygame needs just a little push to make it into conda-forge.

**Comment by richardliaw** [comment]
Yes, skip the ray-rllib package.

**Comment by espentrydal** [comment]
Is support for ppc64le put on? We are setting up a cluster with AC922s and would like to try out Ray if it is feasible.

**Comment by h-vetinari** [comment]
The migrator has chewed through all the dependencies and we've now had #92 & #95. Those are still stuck on compilation issues. Anyone's welcome to try to debug the recipe for that (clone the PR, open a new one based on it, iterate on top).

**Comment by bdice** [comment]
I [tried to get the bot to open a new arch migrator PR](https://github.com/conda-forge/ray-packages-feedstock/pull/182#issuecomment-2688719304) for `ray-packages`, hoping that we might be able to try for this again. However, the migrator status says `ray-packages` is "Awaiting parents." https://conda-forge.org/status/migration/?name=aarch64andppc64leaddition

I wanted to figure out what dependency was blocking the bot from moving forward, so I tried to install an environment with all the dependencies specified in `ray-packages` `meta.yaml`. The only failure was for `vllm`.

```
PackagesNotFoundError: The following packages are not available from current channels:

  - vllm[version='>=0.7.2']                                                                                                                                                                                                                                                                                                                                                                                                         

Current channels:                                                                                                                                                                                                                                                                                                                                                                                                                     

  - https://conda.anaconda.org/conda-forge/linux-aarch64
```

I see that @mattip added `vllm` as a dependency in the conda recipe for `ray-llm` in #198, though it was patched out of the source dependencies -- so not sure what that means.

Where is that `vllm` dependency coming from? I don't see it published in conda-forge. https://anaconda.org/search?q=vllm

I believe the missing `vllm` dependency might be blocking a bot action that would open a new PR attempting to add ARM support to `ray-packages`.

**Comment by zhe-thoughts** [comment]
@pcmoritz What do you think? Thanks

**Comment by carterbox** [comment]
It shouldn't be possible to build and publish a package with off-channel dependencies anymore. I know because I tried with nvmath-python to depend on `pynvjitlink` from `-c rapidsai` and the conda admin blocked the package from upload.

**Comment by carterbox** [comment]
This recipe doesn't test that `ray-llm` (the meta-package that depends on `vllm`) is installable, so the build passes without needing to add an external channel.

**Comment by mattip** [comment]
Sorry, I should have added a test. I see vllm-project/vllm#3126 is still open, and vllm is not available on conda-forge. We should remove the package.

**Comment by bdice** [comment]
#199 removes the vllm dependency. Thanks @mattip!

**Comment by pentschev** [comment]
The aarch64 support has been addressed by #271 .
